### PR TITLE
SM-353 Database and URLs logic change for Device Settings Powertrain

### PR DESCRIPTION
**Pull request by lauradeng23**
# Proposed Changes

### CODE REVIEW

**Comment by lauradeng23** [comment]
Some ideas for how to get device settings only via powertrain:
Idea 1: Change GetDeviceSettingsByTemplate controller: 
- if template_name is not null, fetch device settings via template name as usual
- if template_name is null, query on powertrain and find name with "default" 
Idea 2: Change getConfigURLs function:
- if no matched template, we query for device settings with powertrain (ideally) and names with "default" and takes the first match

**Comment by lauradeng23** [comment]
Didn't fix linter or tests yet, thought I'd get feedback first before going on to fix all that

File: pkg/grpc/device_settings.proto [hunk 1]
```diff
@@ -7,21 +7,21 @@ option go_package = "github.com/DIMO-Network/vehicle-signal-decoding/pkg/grpc";
 package grpc;
 
 message DeviceSettings {
-  string template_name = 1;
```
Comment (JamesReate) [inline pkg/grpc/device_settings.proto:10]: let's still keep the template name, but position 2

File: pkg/grpc/device_settings.proto [hunk 2]
```diff
@@ -7,21 +7,21 @@ option go_package = "github.com/DIMO-Network/vehicle-signal-decoding/pkg/grpc";
...
 message GetDeviceSettingListRequest {
-  optional string template_name = 1;
+  optional string name = 1;
 }
 
 message GetDeviceSettingListResponse {
   repeated DeviceSettings device_settings = 1;
 }
 
 message GetDeviceSettingByTemplateNameRequest {
```
Comment (JamesReate) [inline pkg/grpc/device_settings.proto:23]: ByName

**Comment by JamesReate** [comment]
before deploying this:
- delete all device-settings
after deploy success:
- run seed.sql portion for device-settings (insert rules for each powertrain)

**Comment by JamesReate** [comment]
before deploying this:
- delete all device-settings
after deploy success:
- run seed.sql portion for device-settings (insert rules for each powertrain)

**Pull request merged**